### PR TITLE
DYS report for GLT-2150

### PR DESCRIPTION
**Pull request by callunity**
putting it onto GH for posterity

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Main.java#L70](https://github.com/oicr-gsi/pinery-reports/blob/2cf4003b05394de0321cf10afdba42f97aed4e6e/src/main/java/ca/on/oicr/pineryreports/Main.java#L70): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)
1. ![MAJOR][MAJOR] [Main.java#L72](https://github.com/oicr-gsi/pinery-reports/blob/2cf4003b05394de0321cf10afdba42f97aed4e6e/src/main/java/ca/on/oicr/pineryreports/Main.java#L72): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: docs/DysReport.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Generate
+
+CSV (default format):
+```
+java -jar pinery-reports-<version>-jar-with-dependencies.jar -s <pinery-url> -r dys -o <output-filename>
+```
+
+## Example
+
+| External.Identifier | Stock.Alias               | Aliquot.Alias           | Library.Alias            |
```
Comment (djcooke) [inline docs/DysReport.md]: Table is broken

File: src/main/java/ca/on/oicr/pineryreports/reports/impl/DysReport.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+	  new ColumnDefinition("Targeted.Sequencing"),
+	  new ColumnDefinition("Tissue.Material"),
+	  new ColumnDefinition("Group.ID"),
+	  new ColumnDefinition("Pool.Alias"),
+	  new ColumnDefinition("Num.Dilutions.In.Pool"),
+	  new ColumnDefinition("Pool.Date.Created"),
+	  new ColumnDefinition("Freezer.Box.position")
+  ));
+  
+  private static String DYS = "DYS";
```
[the remark this answers is not in the document]
Comment (djcooke) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/DysReport.java]: should be final, and then Sonar will expect caps

File: src/main/java/ca/on/oicr/pineryreports/reports/impl/DysReport.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    rows.sort(byPoolDate);
+    reportData = rows;
+  }
+  
+  /**
+   * Sort descending by pool date
+   */
+  private final Comparator<ReportObject> byPoolDate = (o1, o2) -> {
+    String o1Created = removeTime(o1.getLane().getPoolCreated());
+    String o2Created = removeTime(o2.getLane().getPoolCreated());
```
Comment (djcooke) [inline src/main/java/ca/on/oicr/pineryreports/reports/impl/DysReport.java]: Probably doesn't matter, but any reason to specifically sort by date and not time?

Comment (callunity) [reply src/main/java/ca/on/oicr/pineryreports/reports/impl/DysReport.java]: Not especially. Will update.

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Main.java#L70](https://github.com/oicr-gsi/pinery-reports/blob/4dff367ea3bd560ca8cf9393a7d649df024bb330/src/main/java/ca/on/oicr/pineryreports/Main.java#L70): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)
1. ![MAJOR][MAJOR] [Main.java#L72](https://github.com/oicr-gsi/pinery-reports/blob/4dff367ea3bd560ca8cf9393a7d649df024bb330/src/main/java/ca/on/oicr/pineryreports/Main.java#L72): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**